cachecontrol/controller.py

```diff
@@ -117,6 +117,8 @@ def cached_request(self, request):
             return resp
 
         headers = CaseInsensitiveDict(resp.headers)
+        if not headers or 'date' not in headers:
+            return False
```

Comment:
Should this do the same check as down below? Check if there is an ``ETag`` and if there isn't one delete this item from the cache? Without a date or an etag this response will never be used for anything.